Add `generateSecretKeySeed()` and `decodeSecretKeySeed()` (#8)
digitalbazaar/bnid: README.md

- `fixedBitLength`: fixed output bit length or 0 to base on input byte size.
  (default: 0)
- `multibase`: `true` to use multibase encoding. (default: `true`)
- `multihash`: `true` to use multihash encoding. (default: `false`)

#### `encode(bytes)`

Encode id bytes into a string.

### `IdDecoder`

> An `IdDecoder` decodes a specific encoding into array of id bytes.

Reviewer: ```suggestion
An `IdDecoder` decodes a specific encoding into an array of bytes representing an ID.
```

Author: Updated 1fe544e

  - Same options as for `IdEncoder`.
- `fixedBitLength`: fixed output bit length.  (default: none)
- `multibase`: `true` to use multibase encoding to detect id format. (default:
  `true`)
- `multihash`: `true` to use multihash encoding. (default: `false`)

> - `expectedSize`: Expected digest size. (default: 32)

Reviewer: ```suggestion
- `expectedSize`: Expected size for multihash-encoded ID bytes. (default: 32)
```

Author: Updated 1fe544e

  - Same options as for `IdEncoder`.
- `fixedBitLength`: fixed output bit length.  (default: none)
- `multibase`: `true` to use multibase encoding to detect id format. (default:
  `true`)
- `multihash`: `true` to use multihash encoding. (default: `false`)

> - `expectedSize`: Expected size for multihash-encoded ID bytes. (default: 32)

Reviewer 2: Document how to skip check.
```suggestion
- `expectedSize`: Expected size for multihash-encoded ID bytes. Use `0` to
  disable size check. (default: 32)
```

Author: Added 075e0a3

- `bitLength`: Number of id bits. (default: 128)
- `multibase`: Account for multibase encoding. (default: true)

> ### `generateMultibaseSeed()`

Reviewer 3: Add example usage code, for these two new entries.

Author: Added 09c7750

- `bitLength`: Number of id bits. (default: 128)
- `multibase`: Account for multibase encoding. (default: true)

### `generateKeySeed(options)`

`generateKeySeed()` and `decodeKeySeed()` methods are for creating  and

> decoding app ids.

Reviewer 3: ```suggestion
decoding key pair seeds which will be used for auto-generating app ids.
```

- `bitLength`: Number of id bits. (default: 128)
- `multibase`: Account for multibase encoding. (default: true)

> ### `generateKeySeed(options)`

Reviewer: Thinking about this more, let's call this `generateSecretKeySeed` to make sure people may be more inclined to be careful with this. Same goes for `decodeSecretKeySeed`.

Author: Renamed .

> `generateKeySeed()` and `decodeKeySeed()` methods are for creating  and
> decoding key pair seeds which will be used for auto-generating app ids.

Reviewer: ```suggestion
`generateKeySeed()` and `decodeKeySeed()` methods are for creating and
decoding secret key pair seeds which can be used to generate public key
based identifiers.
```

Author: Updated bcc742c

> `generateKeySeed()` generates a string key seed ID.

Reviewer: A seed is not an identifier, it's used to generate one.

```suggestion
`generateKeySeed()` generates a secret key seed encoded as a string that
can be stored and later used to generate a key pair. The public key from
the key pair can be used as an identifier.
```

Author: Updated bcc742c

```js
import {generateKeySeed} from 'bnid';
const keySeed = await generateKeySeed();
// Example keySeed: z1Aaj5A4UCsdMpXwdYAReXa4bxWYiKJtdAvB1zMzCHtCbtD
```
Options:
- `encoding`: Encoding. (default: `base58`)
- `bitLength`: Number of id bits. (default: 32 * 8)
- `multibase`: Account for multibase encoding. (default: true)
- `multihash`: Account for multihash encoding. (default: true)

### `decodeKeySeed(options)`

> Decodes a key seed ID into a 32-bytes array seed bytes.

Reviewer: ```suggestion
Decodes an encoded secret key seed into an array of secret key seed bytes (default size: 32 bytes).
```

Author: Updated bcc742c

```js
import {decodeKeySeed} from 'bnid';
const keySeed = 'z1Aaj5A4UCsdMpXwdYAReXa4bxWYiKJtdAvB1zMzCHtCbtD';

> decoded = decodeKeySeed({id: keySeed});

Reviewer: I think conflating secret key seed bytes with an "identifier" is dangerous. We should adjust param names and code to ensure that those two are not mixed. Ultimately, these helper functions are about encoding/decoding secret key seeds for use in identifier generation -- but this library doesn't do the identifier generation, some external library that consumes the seed does (typically, it generates a key pair and uses the public key as the ID or as part of the ID).

The docs should be more clear about this pattern and how this library helps generate secret key bytes and encode them for storage and later generation/regeneration of the same key pair ... *from which* an identifier can be created.

Reviewer: This is also different from the use case we have where we want to generate (typically) 16 random bytes for an identifier and encode it using multibase+multihash.

- `bitLength`: Number of id bits. (default: 128)
- `multibase`: Account for multibase encoding. (default: true)

### `generateSecretKeySeed(options)`

`generateSecretKeySeed()` and `decodeSecretKeySeed()` methods are for creating
and decoding secret key pair seeds which can be used to generate public key
based identifiers.

`generateSecretKeySeed()` generates a secret key seed encoded as a string that
can be stored and later used to generate a key pair. The public key from

> the key pair can be used as an identifier.

Reviewer: ```suggestion
the key pair can be used as an identifier. The encoded key seed MUST be kept secret.
```

Author: Updated 1fe544e

```js
import {generateSecretKeySeed} from 'bnid';
const secretKeySeed = await generateSecretKeySeed();
// Example secretKeySeed: z1Aaj5A4UCsdMpXwdYAReXa4bxWYiKJtdAvB1zMzCHtCbtD
```
Options:
- `encoding`: Encoding. (default: `base58`)
- `bitLength`: Number of id bits. (default: 32 * 8)
- `multibase`: Account for multibase encoding. (default: true)
- `multihash`: Account for multihash encoding. (default: true)

### `decodeSecretKeySeed(options)`

Decodes an encoded secret key seed into an array of secret key seed bytes

> (default size: 32 bytes).

Reviewer: ```suggestion
(default size: 32 bytes). Both the encoded key seed and the decoded bytes MUST be kept secret.
```

Author: Updated 1fe544e

import {generateSecretKeySeed} from 'bnid';
const secretKeySeed = await generateSecretKeySeed();
// Example secretKeySeed: z1Aaj5A4UCsdMpXwdYAReXa4bxWYiKJtdAvB1zMzCHtCbtD
```
Options:
- `encoding`: Encoding. (default: `base58`)
- `bitLength`: Number of id bits. (default: 32 * 8)
- `multibase`: Account for multibase encoding. (default: true)
- `multihash`: Account for multihash encoding. (default: true)

### `decodeSecretKeySeed(options)`

Decodes an encoded secret key seed into an array of secret key seed bytes
(default size: 32 bytes). Both the encoded key seed and the decoded bytes MUST
be kept secret.

```js
import {decodeSecretKeySeed} from 'bnid';
const secretKeySeed = 'z1Aaj5A4UCsdMpXwdYAReXa4bxWYiKJtdAvB1zMzCHtCbtD';
decoded = decodeSecretKeySeed({secretKeySeed});
// Example decoded:
// Uint8Array(32) [
//    80, 174,  15, 131, 124,  59,   9,  51,
//   145, 129,  92, 157, 157, 172, 161,  79,
//    74,  61, 152, 152,  48, 151,  20,  89,
//   225, 169,  71,  34,  49,  61,  21, 215
// ]
```
Options:
- `secretKeySeed`: The secret key seed to be decoded.
- `multibase`: Account for multibase encoding. (default: true)
- `multihash`: Account for multihash encoding. (default: true)

> - `expectedSize`: Expected size for multihash-encoded ID bytes. (default: 32)

Reviewer 2: Document how to skip size check.
```suggestion
- `expectedSize`: Expected size for multihash-encoded ID bytes. Use `0` to
  disable size check. (default: 32)
```

Author: Added 075e0a3
